internal/core/legacy/endpoints/label.go

```diff
@@ -190,5 +234,41 @@ func (e *Endpoints) ListLabelByIDs(ctx context.Context, r *http.Request, vars ma
 		return errorresp.ErrResp(err)
 	}
 
+	projectIDMap := make(map[uint64]struct{})
+	for _, label := range labels {
+		projectIDMap[label.ProjectID] = struct{}{}
+	}
+	for id := range projectIDMap {
+		if err := e.checkProjectPermission(identityInfo, id, apistructs.GetAction); err != nil {
+			return errorresp.ErrResp(err)
```

Comment:
maybe it is enough to judge whether it is an internal-client call at the beginning of the function, this interface is not open to openapi